### PR TITLE
Fix remix simulator transaction endpoints

### PR DESCRIPTION
**Pull request by yann300**
remix-simulator isn't returning a complete (or wrong) data from a few endpoint. 
That causes ethers.js to malfunction (unable to listen on solidity events).

This PR aim to add the missing data and to fix wrong data (nonce, block number).

### CODE REVIEW

File: libs/remix-simulator/src/methods/transactions.ts [hunk 1]
```diff
@@ -95,17 +95,19 @@ export class Transactions {
...
 
-      const txBlock = this.vmContext.txs[receipt.hash]
+      const txBlock = this.vmContext.blockByTxHash[receipt.hash]
+
+      const logs = this.vmContext.logsManager.getLogsByTxHash(receipt.hash)
 
       const r: Record <string, unknown> = {
         transactionHash: receipt.hash,
-        transactionIndex: '0x00',
+        transactionIndex: '0x0',
```
Comment (Aniket-Engg) [inline libs/remix-simulator/src/methods/transactions.ts:104]: can this be hardcoded for VM?

File: package.json [hunk 2]
```diff
@@ -74,7 +74,7 @@
     "nightwatch_local_specialFunctions": "npm run build:e2e && nightwatch --config dist/apps/remix-ide-e2e/nightwatch.js dist/apps/remix-ide-e2e/src/tests/specialFunctions.test.js --env=chrome",
     "nightwatch_local_solidityUnitTests": "npm run build:e2e && nightwatch --config dist/apps/remix-ide-e2e/nightwatch.js dist/apps/remix-ide-e2e/src/tests/solidityUnittests.spec.js --env=chrome",
     "nightwatch_local_remixd": "npm run build:e2e && nightwatch --config dist/apps/remix-ide-e2e/nightwatch.js dist/apps/remix-ide-e2e/src/tests/remixd.test.js --env=chrome",
-    "nightwatch_local_terminal": "npm run build:e2e && nightwatch --config dist/apps/remix-ide-e2e/nightwatch.js dist/apps/remix-ide-e2e/src/tests/terminal.test.js --env=chrome",
+    "nightwatch_local_terminal": "npm run build:e2e && nightwatch --config dist/apps/remix-ide-e2e/nightwatch.js dist/apps/remix-ide-e2e/src/tests/terminal.test.js --env=firefox",
```
Comment (Aniket-Engg) [inline package.json:77]: is this change required?

File: libs/remix-debug/src/storage/storageResolver.ts [hunk 3]
```diff
@@ -133,7 +133,7 @@ export class StorageResolver {
         resolve([{}, null])
       } else {
         this.web3.debug.storageRangeAt(
```
Comment (Aniket-Engg) [inline libs/remix-debug/src/storage/storageResolver.ts:135]: does this require change in remix-debug unit tests?

**Review by Aniket-Engg — COMMENTED** [review]
Not sure if unit tests corresponding to libs changes exist or should be updated

File: libs/remix-debug/src/storage/storageResolver.ts (hunk 3 above)
Comment (yann300) [reply libs/remix-debug/src/storage/storageResolver.ts:135]: I don't believe so, this code should already be touched by remix-debug unit test.

**Review by Aniket-Engg — APPROVED** [review]
